offdispmap/ContentViewiOS.swift

```diff
@@ -0,0 +1,226 @@
+                Button(action: {
+                    selectDispensary(dispensary)
+                    selectedDetent = .fraction(0.25)
+                }) {
+                    DispensaryRow(dispensary: dispensary, isSelected: dispensary.id == selectedDispensary?.id)
+                        .contentShape(Rectangle())
+                }
+                .buttonStyle(PlainButtonStyle())
+                Divider()
+
+            }.listStyle(.plain)
+        }
+    }
+    
+    var body: some View {
+            MapView(
+                annotations: dispensaryAnnotations,
+                selectedAnnotation: selectedAnnotation,
+                annotationFilter: { annotation in
+                    (self.nycOnlyMode ? annotation.dispensary.isNYC : true)
+                },
+                onAnnotationSelect: { annotation in
+                    selectDispensary(annotation.dispensary)
+                })
+            .ignoresSafeArea()
+            .sheet(isPresented: .constant(true)) {
+                ScrollView {
+                    VStack {
+                        HStack {
+                            Text(headerTitle)
+                                .font(.title3)
+                                .fontWeight(.bold)
+                            Spacer()
+                        }
+                        .padding(.horizontal)
+                        .padding(.top)
+                        
+                        statisticsView
+                            .padding(.horizontal)
+                        Divider()
+                        
+                        dispensaryListView
+                            .padding(.horizontal)
+                            .frame(maxWidth: .infinity, maxHeight: .infinity)
+                            .presentationDetents([.fraction(0.25), .medium], selection: $selectedDetent)
+                            .presentationBackgroundInteraction(.enabled)
+                            .interactiveDismissDisabled()
+                            .presentationBackground(.thinMaterial)
+                    }
+                }
+            }
+            .onAppear {
+                if displayDispensaries.isEmpty { // So the preview works
+                    reload()
+                }
+            }
+            .onChange(of: nycOnlyMode) {
+                reload()
+            }
+            .onChange(of: deliveryOnlyMode) {
+                reload()
+            }
+    }
+    
+    private func selectDispensary(_ dispensary: Dispensary) {
+        selectedDispensary = dispensary
+        if dispensary.isTemporaryDeliveryOnly {
+            // This is currently not possible due to the UI code,
+            // however this case should be gracefully handled probably?
+            // Maybe I can refactor this out of existence
+            return;
+        }
+        if dispensary.coordinate == nil {
+            // This should be rare, but its when we loaded
```

Comment:
FWIW I've worked around this before. 
I pushed geocoding onto a background task that ran on a throttled loop that more slowly updated the cache. 
it's a bit different in this app since the pins are a core part of the UX and displayed on launch. But a little pop-in on the first launch probably won't hurt anyone.

You could also consider mirroring the dataset and re-hosting it with the coordinates already attached. (Github pages/static file, Cloudflare, S3, etc would make this somewhere between free and very, very cheap)